NGitLab.Mock/MergeRequest.cs

```diff
@@ -380,8 +381,13 @@ private void RefreshInternalState()
             _divergedCommitsCount = Project.Repository.ComputeDivergence(TargetBranchHeadCommit, commonCommit);
 
             // Determine if there are conflicts, by performing a rebase on a transient copy of the source branch
-            var transientBranchName = "transient/" + Guid.NewGuid().ToString("N");
-            Project.Repository.CreateBranch(transientBranchName, _consolidatedSourceBranch);
+            var smallBranchName = Guid.NewGuid().ToString("N");
+            var transientBranchName = "refs/transient/" + smallBranchName;
+
+            var branchFullPath = Path.Combine(Project.Repository.FullPath, ".git/refs/transient", smallBranchName);
```

Comment:
We are using Libgit2sharp in the repo. So, I think you can use `repo.Refs.Add(transientBranchName, ...)`. Make you can need to add an internal method in `Project.Repository` or expose a way to get the LibGit2Sharp repository object.

Reply:
I can add a ref with LibGit2Sharp but this ref is not usable for rebase. I tried an other way, I use the merge without committing to check conflicts and I abort operation after that. See : https://stackoverflow.com/a/501461